pipeline-in-pod/pkg/reconciler/pipelineinpod/pipelineinpod.go

```diff
@@ -258,3 +271,79 @@ func (c *Reconciler) getTaskSpecs(cpr *cprv1alpha1.ColocatedPipelineRunStatus) (
 	}
 	return tasks, nil
 }
+
+func (r *Reconciler) getPodForColocatedPipelineRun(ctx context.Context, cpr *cprv1alpha1.ColocatedPipelineRun) (*corev1.Pod, error) {
+	logger := logging.FromContext(ctx)
+	labelSelector := fmt.Sprintf("%s=%s", cprv1alpha1.ColocatedPipelineRunLabelKey, cpr.Name)
+	pods, err := r.kubeClientSet.CoreV1().Pods(cpr.Namespace).List(ctx, metav1.ListOptions{
```

Comment:
fine for experimental, but we should probably switch to a lister instead of making direct API server calls